spec/components/provider_interface/course_details_component_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'rails_helper'
+
+RSpec.describe ProviderInterface::CourseDetailsComponent do
+  let(:application_choice) {
+    create(:application_choice,
+           course: create(:course, funding_type: 'fee'))
+  }
+
+  let(:application_choice_with_accredited_body) {
+    create(:application_choice,
+           course: create(:course, :accredited_provider))
+  }
+
+  let(:render) { render_inline(described_class.new(application_choice: application_choice)) }
+
+  it 'renders the provider name and code' do
+    expect(render.css('.govuk-summary-list__row')[0].text).to include('Provider')
+    expect(render.css('.govuk-summary-list__row')[0].text).to include(application_choice.provider.name_and_code)
+  end
+
+  context 'when an accredited body is present' do
+    it 'renders the accredited body name and code when it is present' do
+      render = render_inline(described_class.new(application_choice: application_choice_with_accredited_body))
+      expect(render.css('.govuk-summary-list__row')[1].text).to include('Accredited body')
+      expect(render.css('.govuk-summary-list__row')[1].text).to include(application_choice_with_accredited_body.course.accredited_provider.name)
+    end
+  end
+
+  context 'when an accredited body is not present' do
+    it 'renders the provider name and code in place of the accredited body name and code' do
+      expect(render.css('.govuk-summary-list__row')[1].text).to include('Accredited body')
+      expect(render.css('.govuk-summary-list__row')[1].text).to include(application_choice.provider.name_and_code)
+    end
+  end
+
+  it 'renders the course name and code' do
+    expect(render.css('.govuk-summary-list__row')[2].text).to include('Course')
+    expect(render.css('.govuk-summary-list__row')[2].text).to include(application_choice.course.name_and_code)
+  end
+
+  it 'renders the recruitment cycle year' do
+    expect(render.css('.govuk-summary-list__row')[3].text).to include('Cycle')
+    expect(render.css('.govuk-summary-list__row')[3].text).to include(application_choice.course.recruitment_cycle_year.to_s)
+  end
+
+  it 'renders the preferred location' do
+    expect(render.css('.govuk-summary-list__row')[4].text).to include('Preferred location')
+    expect(render.css('.govuk-summary-list__row')[4].text).to include(application_choice.site.name_and_code)
+  end
+
+  it 'renders the study mode' do
+    expect(render.css('.govuk-summary-list__row')[5].text).to include('Study mode')
+    expect(render.css('.govuk-summary-list__row')[5].text).to include(application_choice.course_option.study_mode.humanize)
+  end
+
+  it 'renders financing funding type of a course' do
+    expect(render.css('.govuk-summary-list__row')[6].text).to include('Funding type')
+    expect(render.css('.govuk-summary-list__row')[6].text).to include('Fee paying')
+  end
```

Comment:
I appreciate the thoroughness of this testing!

Two questions:

- When we make changes in the ordering, we would need to change a lot of the assertions. Can we avoid that somehow?
- Some of the assertions duplicate the code that's being tested. For example, the study mode test in essence tests that `application_choice.course_option.study_mode.humanize` is the same as `application_choice.course_option.study_mode.humanize`. Perhaps you could make it specific, so we test that the resulting study mode string actually makes sense (and isn't a sneaky nil for example).

Reply:
have added a method that looks rows up by name and returns an index to make it easier to change row orders in future (they are now all in one place).

Reply:
I have added literals to the test now 👍

Reply:
Also moved to instance_doubles as there is no need for DB creation with this kind of component test in my mind.